### PR TITLE
set etag and permission fields for trashbin entries

### LINKED ISSUE #21378: removing files from trash: Undefined index: permissions at /var/www/owncloud/lib/private/files/fileinfo.php#177
Hi,

removing files from trash results in the following, reproducible error:

Undefined index: permissions at /var/www/owncloud/lib/private/files/fileinfo.php#177 

Files get successfully deleted.

file permissions:
root@server:/var/www/owncloud/data/testuser/files_trashbin/files# ls -al
insgesamt 40
drwxr-xr-x 2 www-data www-data  4096 Dez 27 21:18 .
drwxr-xr-x 5 www-data www-data  4096 Dez 27 17:42 ..
-rw-r--r-- 1 www-data www-data 29055 Dez 27 21:18 testfile.txt.d1451247492

sudo -u www-data php ./occ app:list
Enabled:
  - activity: 2.1.3
  - calendar: 0.8.1
  - contacts: 0.5.0.0
  - direct_menu: 0.5
  - files: 1.2.0
  - files_external: 0.3.0
  - files_pdfviewer: 0.7
  - files_sharing: 0.7.0
  - files_texteditor: 2.0
  - files_trashbin: 0.7.0
  - files_versions: 1.1.0
  - files_videoviewer: 0.1.3
  - firstrunwizard: 1.1
  - gallery: 14.2.0
  - provisioning_api: 0.3.0
  - templateeditor: 0.1
  - user_otp: 2.5

Disabled:
  - documents
  - encryption
  - external
  - notifications
  - ownbackup
  - storagecharts2
  - user_external
  - user_ldap

System:
Owncloud:  8.2.2 [ http://download.owncloud.org/download/repositories/stable/Ubuntu_14.04/ ]
System: Ubuntu 14.04 LTS
Webserver: apache2 2.4.7-1ubuntu4.8
PHP: php5 5.5.9+dfsg-1ubuntu4.14
Database: mysql 5.5.46-0ubuntu0.14.04.2

Ciao
Marcus

### PR DESCRIPTION
**Pull request by icewind1991**
Properly set all the fields that are expected

Fixes #21378 

cc @PVince81 @schiesbn

### CODE REVIEW

**Comment by PVince81** [comment]
:+1: as discussed

File: apps/files_trashbin/lib/helper.php [hunk 1]
```diff
@@ -91,6 +92,8 @@ public static function getTrashFiles($dir, $user, $sortAttribute = '', $sortDesc
 						'type' => $view->is_dir($dir . '/' . $entryName) ? 'dir' : 'file',
 						'directory' => ($dir === '/') ? '' : $dir,
 						'size' => $size,
+						'etag' => '',
+						'permissions' => Constants::PERMISSION_ALL - Constants::PERMISSION_SHARE
```
Comment (rullzer) [inline apps/files_trashbin/lib/helper.php]: Files can't have create permissions.

Comment (rullzer) [inline apps/files_trashbin/lib/helper.php]: So nothing bad really happens probably. But for cleanness.

**Comment by PVince81** [comment]
```
{... ,"app":"PHP","message":"Class 'OCA\Files_Trashbin\Constants' not found at \/var\/www\/owncloud\/apps\/files_trashbin\/lib\/helper.php#95","level":3,"time":"2016-01-10T21:42:55+01:00"}
```

**Comment by icewind1991** [comment]
> `+use OCP\Constants;`

...

**Comment by PVince81** [comment]
Yep, works indeed.

**Comment by DeepDiver1975** [comment]
:+1:

**Comment by MorrisJobke** [comment]
@karlitschek @icewind1991 Could we backport this to stable8.1 and stable8.2?

**Comment by karlitschek** [comment]
i think a backport makes sense :+1:

**Comment by MorrisJobke** [comment]
stable8.2 #22571 
stable8.1 #22572

**Pull request merged**